dev/core/src/com/google/gwt/core/ext/linker/impl/installScriptEarlyDownload.js

```diff
@@ -19,23 +19,15 @@ function installScript(filename) {
     var doc = getInstallLocationDoc();
     var docbody = doc.body;
     var script;
-    // for sourcemaps, we inject textNodes into the script element on Chrome
-    if (navigator.userAgent.indexOf("Chrome") > -1 && window.JSON) {
-      var scriptFrag = doc.createDocumentFragment()
-      // surround code with eval until crbug #90707 
-      scriptFrag.appendChild(doc.createTextNode("eval(\""));
+    // for sourcemaps, we inject the code as a single string for Chrome
+    if (navigator.userAgent.indexOf("Chrome") > -1) {
+      var codeString = "";
       for (var i = 0; i < code.length; i++) {
-        // escape newlines, backslashes, and quotes with JSON.stringify
-        // rather than create multiple script tags which mess up line numbers, we use 1 tag, multiple text nodes
-        var c = window.JSON.stringify(code[i]); 
-        // trim beginning/end quotes
-        scriptFrag.appendChild(doc.createTextNode(c.substring(1, c.length - 1)));
+        codeString += code[i];
```

Comment:
Why not simply `script.text = code.join();`?

Reply:
TBH I mostly looked at the recent answers on https://stackoverflow.com/questions/7296594/array-join-vs-string-concat

Reply:
Methodology matters a lot - remember for the answers that do loop(`push`)+`join` and compare to loop(`+=`) that we don't even have a loop - the array already exists, we're only doing `join` and comparing it against loop(`+=`). https://stackoverflow.com/a/65402300/860630 seems to test this directly, while https://stackoverflow.com/a/67840218/860630 tests the combined "make the array" and "join it"). I'm not sure there's a clear winner in that light?

That said, if it is close, I'd personally vote for concise/clarity.

Reply:
> Why not simply `script.text = code.join();`?

+1

Use `.join("")` though, as `.join()` without argument uses an implicit `","` separator

Reply:
Thanks for the pointers, moved to `join('')`